patches/MCLOUD-5837__fix_filesystem_load_balancer_issue.patch

```diff
@@ -0,0 +1,152 @@
+      */
+     const MAX_IMAGE_HEIGHT = 1200;
+
+@@ -169,21 +170,32 @@ class Uploader
+      */
+     private $directoryList;
+
++    /**
++     * @var DriverPool|null
++     */
++    private $driverPool;
++
++    /**
++     * @var DriverInterface|null
++     */
++    private $fileDriver;
++
+     /**
+      * Init upload
+      *
+      * @param string|array $fileId
+      * @param \Magento\Framework\File\Mime|null $fileMime
+      * @param DirectoryList|null $directoryList
++     * @param DriverPool|null $driverPool
+      * @throws \DomainException
+      */
+     public function __construct(
+         $fileId,
+         Mime $fileMime = null,
+-        DirectoryList $directoryList = null
++        DirectoryList $directoryList = null,
++        DriverPool $driverPool = null
+     ) {
+-        $this->directoryList= $directoryList ?: \Magento\Framework\App\ObjectManager::getInstance()
+-            ->get(DirectoryList::class);
++        $this->directoryList= $directoryList ?: ObjectManager::getInstance()->get(DirectoryList::class);
+
+         $this->_setUploadFileId($fileId);
+         if (!file_exists($this->_file['tmp_name'])) {
+@@ -192,7 +204,8 @@ class Uploader
+         } else {
+             $this->_fileExists = true;
+         }
+-        $this->fileMime = $fileMime ?: \Magento\Framework\App\ObjectManager::getInstance()->get(Mime::class);
++        $this->fileMime = $fileMime ?: ObjectManager::getInstance()->get(Mime::class);
++        $this->driverPool = $driverPool;
+     }
+
+     /**
+@@ -230,7 +243,7 @@ class Uploader
+             $this->setAllowCreateFolders(true);
+             $this->_dispretionPath = static::getDispersionPath($fileName);
+             $destinationFile .= $this->_dispretionPath;
+-            $this->_createDestinationFolder($destinationFile);
++            $this->createDestinationFolder($destinationFile);
+         }
+
+         if ($this->_allowRenameFiles) {
+@@ -275,13 +288,11 @@ class Uploader
+      * @return void
+      * @throws FileSystemException
+      */
+-    private function validateDestination($destinationFolder)
++    private function validateDestination(string $destinationFolder): void
+     {
+         if ($this->_allowCreateFolders) {
+-            $this->_createDestinationFolder($destinationFolder);
+-        }
+-
+-        if (!is_writable($destinationFolder)) {
++            $this->createDestinationFolder($destinationFolder);
++        } elseif (!$this->getFileDriver()->isWritable($destinationFolder)) {
+             throw new FileSystemException(__('Destination folder is not writable or does not exists.'));
+         }
+     }
+@@ -655,7 +666,7 @@ class Uploader
+      * @return \Magento\Framework\File\Uploader
+      * @throws FileSystemException
+      */
+-    private function _createDestinationFolder($destinationFolder)
++    private function createDestinationFolder(string $destinationFolder)
```

Comment:
Missing return type after prototype declaration

Reply:
This patch is based on code changes that are delivering to the mainline. So it's better to keep changes as is.

Reply:
Okay that's acceptable